src/iterator.cr

```diff
@@ -99,18 +99,33 @@ module Iterator(T)
     self
   end
 
+  # Returns a new array with the results of running block once for
+  # every element in enum.
+  #
+  # ```
+  # [1, 2, 3].map &.*(2) # => [2, 4, 6]
+  # ```
```

Comment:
Yes, adding a `to_a` call should be enough, but I agree the docs should say "Returns an iterator that ..."